format/format.go

```diff
@@ -255,19 +255,69 @@ func (w *writer) writeRule(rule *ast.Rule, isElse bool, comments []*ast.Comment)
 	w.startLine()
 	w.write("}")
 	if rule.Else != nil {
-		if len(comments) > 0 && comments[0].Location.Row-rule.Else.Head.Location.Row < 0 {
-			w.blankLine()
-		} else {
-			w.write(" ")
-		}
-		rule.Else.Head.Name = ast.Var("else")
-		rule.Else.Head.Args = nil
-		comments = w.insertComments(comments, rule.Else.Head.Location)
-		comments = w.writeRule(rule.Else, true, comments)
+		comments = w.writeElse(rule, comments)
 	}
 	return comments
 }
 
+func (w *writer) writeElse(rule *ast.Rule, comments []*ast.Comment) []*ast.Comment {
+	// If there was nothing else on the line before the "else" starts
+	// then preserve this style of else block, otherwise it will be
+	// started as an "inline" else eg:
+	//
+	//     p {
+	//     	...
+	//     }
+	//
+	//     else {
+	//     	...
+	//     }
+	//
+	// versus
+	//
+	//     p {
+	// 	    ...
+	//     } else {
+	//     	...
+	//     }
+	//
+	// Note: This doesn't use the `close` as it currently isn't accurate for all
+	// types of values. Checking the actual line text is the most consistent approach.
+	wasInline := false
+	ruleLines := bytes.Split(rule.Location.Text, []byte("\n"))
+	relativeElseRow := rule.Else.Location.Row - rule.Location.Row
+	if relativeElseRow > 0 && relativeElseRow < len(ruleLines) {
```

Comment:
Can we explain what's going on here more explicitly in the comment. Maybe it's just me but it would help to do so.